src/rulesets/SparseArrays/sparsematrix.jl

```diff
@@ -10,16 +10,59 @@ function rrule(::typeof(sparse), I::AbstractVector, J::AbstractVector, V::Abstra
     return sparse(I, J, V, m, n, combine), sparse_pullback
 end
 
-function rrule(::Type{T}, A::AbstractMatrix) where T <: SparseMatrixCSC
+function rrule(::Type{T}, A::AbstractMatrix) where T <: AbstractSparseMatrix
     function sparse_pullback(Ω̄)
         return NoTangent(), Ω̄
     end
     return T(A), sparse_pullback
 end
 
-function rrule(::Type{T}, v::AbstractVector) where T <: SparseVector
+function rrule(::Type{T}, v::AbstractVector) where T <: AbstractSparseVector
     function sparse_pullback(Ω̄)
         return NoTangent(), Ω̄
     end
     return T(v), sparse_pullback
 end
+
+function rrule(::typeof(findnz), A::AbstractSparseMatrix)
+    I, J, V = findnz(A)
+    m, n = size(A)
+
+    function findnz_pullback(Δ)
+        Δ === NoTangent() && return (NoTangent(), Δ)
+        Δ === ZeroTangent() && return (NoTangent(), Δ)
+        
```

Comment:
I think these zero cases are supposed to be handled by AD, and need not be written out each time:
```suggestion
    function findnz_pullback(Δ)
```